### ISSUE TITLE
Converter doesn't work for LDC2015E73

### ISSUE THREAD

**Issue by pfllo** [hunterhector/EvmEval#12]
Hi

I'm a participant of TAC 2015 Event Nugget Detection task.
I think the training data for this year is LDC2015E73 (TAC KBP 2015 Event Nugget Training Data Annotation), right?

However, the converter doesn't work this data set. (I tried 'demo_conversion.sh')
The xml file names look like: 1b386c986f9d06fd0a0dda70c3b8ade9.event_nuggets.xml
There is no mpdf file and rich_ere file.
This doesn't get better when I change the file name above to:  1b386c986f9d06fd0a0dda70c3b8ade9.rich_ere.xml

Am I using the wrong data?
If not, can you please tell me how can I evaluate the performance of my system on this dataset?

Thanks a lot.

**Comment by hunterhector** [comment]
Hi, Luo,

Sorry for not replying early enough, I was out of town last week.

So I guess your problem is caused by the new extension deployed by LDC, which is different from a previous dataset we received. We added an option in our converter to specify the file extension now, so that you don't have to do it yourself. 

On the other hand, we current still require all LDC files to be located in one folder. So if you put both event_nuggets.xml and .txt in to the folder, you could follow something similar to shell script below:

https://github.com/hunterhector/EvmEval/blob/master/test_conversion.sh

Let me know if this works. If it fails, it will be easier if we know your steps and the error messages.

**Comment by pfllo** [comment]
It seems you didn't upload your new 'bin/converter-1.0.1-jar-with-dependencies.jar'

It gives following error:

```
Running XML to Brat Converter...
Error: Unable to access jarfile bin/converter-1.0.1-jar-with-dependencies.jar
Running tokenizer...
[INFO] Successfully completed.
Converting to TBF format
[INFO] 2015-08-09 16:03:43,151 : Finish converting 0 files
```

**Comment by hunterhector** [comment]
Woops, sorry, gitignore trick me on that one. Now it should be there.

**Comment by pfllo** [comment]
It gives following warns and errors:

```
Running XML to Brat Converter...
[WARN] Invalid offset 562, 569 of event mention [       ] for string [charded] at doc [bolt-eng-DF-170-181109-47916]
[WARN] Invalid offset 586, 593 of event mention [       ] for string [lashing] at doc [bolt-eng-DF-170-181109-47916]
[WARN] Invalid offset 242, 249 of event mention [       ] for string [charges] at doc [bolt-eng-DF-170-181109-48534]
[WARN] Invalid offset 869, 881 of event mention [stepped
down] for string [stepped down] at doc [XIN_ENG_20090915.0127]
[INFO] Processing document 1b386c986f9d06fd0a0dda70c3b8ade9
[INFO] Processing document 1ef1a80e902c1fc4507ca6285ab740ec
[INFO] Processing document 3288ddfcb46d1934ad453afd8a4e292f
[INFO] Processing document 3f71fead3fa119ccdcdf01769ffee5b1
[INFO] Processing document AFP_ENG_20090801.0002
[INFO] Processing document AFP_ENG_20090806.0680
[INFO] Processing document AFP_ENG_20090814.0225
[INFO] Processing document AFP_ENG_20090818.0072
[INFO] Processing document AFP_ENG_20090827.0649
[INFO] Processing document AFP_ENG_20090906.0009
[INFO] Processing document AFP_ENG_20091004.0162
[INFO] Processing document AFP_ENG_20091005.0318
[INFO] Processing document AFP_ENG_20091008.0600
[INFO] Processing document AFP_ENG_20091015.0364
[INFO] Processing document AFP_ENG_20091021.0463
[INFO] Processing document AFP_ENG_20091102.0449
[INFO] Processing document AFP_ENG_20091109.0045
[INFO] Processing document AFP_ENG_20091109.0326
[INFO] Processing document AFP_ENG_20091118.0674
[INFO] Processing document AFP_ENG_20100107.0147
[INFO] Processing document AFP_ENG_20100130.0284
[INFO] Processing document AFP_ENG_20100407.0477
[INFO] Processing document AFP_ENG_20100423.0583
[INFO] Processing document AFP_ENG_20100428.0342
[INFO] Processing document AFP_ENG_20100505.0537
[INFO] Processing document AFP_ENG_20100513.0269
[INFO] Processing document AFP_ENG_20100630.0660
[INFO] Processing document AFP_ENG_20100718.0338
[INFO] Processing document AFP_ENG_20100729.0356
[INFO] Processing document AFP_ENG_20101106.0133
[INFO] Processing document AFP_ENG_20101228.0209
[INFO] Processing document APW_ENG_20090605.0323
[INFO] Processing document APW_ENG_20090611.0337
[INFO] Processing document APW_ENG_20090611.0436
[INFO] Processing document APW_ENG_20090820.0822
[INFO] Processing document APW_ENG_20090829.0063
[INFO] Processing document APW_ENG_20090914.1248
[INFO] Processing document APW_ENG_20091028.0874
[INFO] Processing document APW_ENG_20091106.0412
[INFO] Processing document APW_ENG_20091208.0983
[INFO] Processing document APW_ENG_20100120.1272
[INFO] Processing document APW_ENG_20100122.0831
[INFO] Processing document APW_ENG_20100124.0007
[INFO] Processing document APW_ENG_20100128.0353
[INFO] Processing document APW_ENG_20100204.0044
[INFO] Processing document APW_ENG_20100304.0736
[INFO] Processing document APW_ENG_20100508.0084
[INFO] Processing document APW_ENG_20100512.0057
[INFO] Processing document APW_ENG_20100629.0013
[INFO] Processing document APW_ENG_20100812.0033
[INFO] Processing document APW_ENG_20100826.0686
[INFO] Processing document APW_ENG_20100909.0028
[INFO] Processing document APW_ENG_20100914.0967
[INFO] Processing document APW_ENG_20100917.0504
[INFO] Processing document APW_ENG_20100918.0030
[INFO] Processing document APW_ENG_20101024.0551
[INFO] Processing document APW_ENG_20101117.0143
[INFO] Processing document APW_ENG_20101125.0441
[INFO] Processing document APW_ENG_20101206.0775
[INFO] Processing document APW_ENG_20101214.0097
[INFO] Processing document bolt-eng-DF-170-181103-8900160
[INFO] Processing document bolt-eng-DF-170-181103-8901874
[INFO] Processing document bolt-eng-DF-170-181103-8908896
[INFO] Processing document bolt-eng-DF-170-181103-8911799
[INFO] Processing document bolt-eng-DF-170-181109-47916
[INFO] Processing document bolt-eng-DF-170-181109-48534
[INFO] Processing document bolt-eng-DF-170-181109-57345
[INFO] Processing document bolt-eng-DF-170-181109-60453
[INFO] Processing document bolt-eng-DF-170-181113-62860
[INFO] Processing document bolt-eng-DF-170-181113-72312
[INFO] Processing document bolt-eng-DF-170-181113-82408
[INFO] Processing document bolt-eng-DF-170-181118-8874957
[INFO] Processing document bolt-eng-DF-170-181122-8780882
[INFO] Processing document bolt-eng-DF-170-181122-8790737
[INFO] Processing document bolt-eng-DF-170-181122-8791540
[INFO] Processing document bolt-eng-DF-170-181122-8793271
[INFO] Processing document bolt-eng-DF-170-181122-8793828
[INFO] Processing document bolt-eng-DF-170-181122-8798940
[INFO] Processing document bolt-eng-DF-170-181122-8803193
[INFO] Processing document bolt-eng-DF-170-181122-8803967
[INFO] Processing document bolt-eng-DF-170-181122-8808533
[INFO] Processing document bolt-eng-DF-170-181122-8818559
[INFO] Processing document bolt-eng-DF-170-181125-9125545
[INFO] Processing document bolt-eng-DF-170-181125-9126447
[INFO] Processing document bolt-eng-DF-170-181130-92738
[INFO] Processing document bolt-eng-DF-170-181130-95269
[INFO] Processing document bolt-eng-DF-170-181137-9031763
[INFO] Processing document bolt-eng-DF-183-195681-7948858
[INFO] Processing document bolt-eng-DF-199-192783-6849980
[INFO] Processing document bolt-eng-DF-199-192783-6850348
[INFO] Processing document bolt-eng-DF-199-192783-6861664
[INFO] Processing document bolt-eng-DF-199-192783-6864512
[INFO] Processing document bolt-eng-DF-199-192821-6708363
[INFO] Processing document bolt-eng-DF-199-192823-6755287
[INFO] Processing document bolt-eng-DF-199-192823-6760150
[INFO] Processing document bolt-eng-DF-199-192909-6666973
[INFO] Processing document bolt-eng-DF-199-193201-6644457
[INFO] Processing document bolt-eng-DF-200-192392-4569357
[INFO] Processing document bolt-eng-DF-200-192403-6250142
[INFO] Processing document bolt-eng-DF-200-192431-6338651
[INFO] Processing document bolt-eng-DF-200-192446-15905846
[INFO] Processing document bolt-eng-DF-200-192446-3807734
[INFO] Processing document bolt-eng-DF-200-192446-3810246
[INFO] Processing document bolt-eng-DF-200-192446-3810611
[INFO] Processing document bolt-eng-DF-200-192448-6191774
[INFO] Processing document bolt-eng-DF-200-192451-5802089
[INFO] Processing document bolt-eng-DF-200-192451-5802600
[INFO] Processing document bolt-eng-DF-200-192453-20201647
[INFO] Processing document bolt-eng-DF-200-192453-5806585
[INFO] Processing document bolt-eng-DF-201-185516-356700
[INFO] Processing document bolt-eng-DF-201-185522-16013565
[INFO] Processing document bolt-eng-DF-203-185926-21042296
[INFO] Processing document bolt-eng-DF-203-185933-21070100
[INFO] Processing document bolt-eng-DF-203-185933-21070959
[INFO] Processing document bolt-eng-DF-203-185933-21070991
[INFO] Processing document bolt-eng-DF-203-185933-458371
[INFO] Processing document bolt-eng-DF-203-185938-16105295
[INFO] Processing document bolt-eng-DF-203-185938-21062038
[INFO] Processing document bolt-eng-DF-203-185938-393068
[INFO] Processing document bolt-eng-DF-203-185938-398283
[INFO] Processing document bolt-eng-DF-203-185938-411334
[INFO] Processing document bolt-eng-DF-203-185938-411582
[INFO] Processing document bolt-eng-DF-203-185940-16087934
[INFO] Processing document bolt-eng-DF-203-185940-16088129
[INFO] Processing document bolt-eng-DF-203-185940-16090644
[INFO] Processing document bolt-eng-DF-203-185940-16091824
[INFO] Processing document bolt-eng-DF-203-185940-21049635
[INFO] Processing document bolt-eng-DF-203-185940-349202
[INFO] Processing document bolt-eng-DF-203-185940-357818
[INFO] Processing document bolt-eng-DF-203-185943-16124082
[INFO] Processing document bolt-eng-DF-212-191665-3129265
[INFO] Processing document bolt-eng-DF-212-191666-3032455
[INFO] Processing document bolt-eng-DF-212-191668-3064026
[INFO] Processing document CNA_ENG_20100417.0023
[INFO] Processing document CNA_ENG_20100420.0043
[INFO] Processing document CNA_ENG_20101001.0032
[INFO] Processing document NYT_ENG_20090707.0014
[INFO] Processing document NYT_ENG_20090825.0142
[INFO] Processing document NYT_ENG_20100216.0142
[INFO] Processing document NYT_ENG_20100428.0093
[INFO] Processing document NYT_ENG_20101117.0179
[INFO] Processing document NYT_ENG_20130607.0076
[INFO] Processing document NYT_ENG_20130628.0102
[INFO] Processing document NYT_ENG_20130712.0134
[INFO] Processing document NYT_ENG_20130816.0069
[INFO] Processing document NYT_ENG_20130822.0211
[INFO] Processing document WPB_ENG_20100917.0028
[INFO] Processing document WPB_ENG_20101228.0045
[INFO] Processing document XIN_ENG_20090618.0341
[INFO] Processing document XIN_ENG_20090707.0358
[INFO] Processing document XIN_ENG_20090915.0127
[INFO] Processing document XIN_ENG_20090915.0142
[INFO] Processing document XIN_ENG_20100114.0378
[INFO] Processing document XIN_ENG_20100206.0090
[INFO] Processing document XIN_ENG_20100318.0109
[INFO] Processing document XIN_ENG_20100508.0093
[INFO] Processing document XIN_ENG_20100613.0159
[INFO] Processing document XIN_ENG_20101006.0324
Running tokenizer...
Adding annotator tokenize
TokenizerAnnotator: No tokenizer type provided. Defaulting to PTBTokenizer.
[WARN] Boundary mismatch found in bolt-eng-DF-170-181103-8911799: Token [18600,18612] [explanations] vs. EventMentionSpan [18600,18611] [explanation]
[WARN] Boundary mismatch found in bolt-eng-DF-170-181103-8911799: Token [18996,19008] [explanations] vs. EventMentionSpan [18996,19007] [explanation]
[WARN] Boundary mismatch found in bolt-eng-DF-170-181103-8911799: Token [19612,19624] [explanations] vs. EventMentionSpan [19612,19623] [explanation]
[WARN] Boundary mismatch found in bolt-eng-DF-170-181122-8803193: Token [10969,10974] [jail.] vs. EventMentionSpan [10969,10973] [jail]
[WARN] Boundary mismatch found in bolt-eng-DF-170-181122-8803193: Token [12577,12582] [jail.] vs. EventMentionSpan [12577,12581] [jail]
[WARN] Boundary mismatch found in bolt-eng-DF-170-181125-9126447: Token [7194,7203] [responses] vs. EventMentionSpan [7194,7202] [response]
[WARN] Boundary mismatch found in bolt-eng-DF-199-192783-6849980: Token [245,252] [antiwar] vs. EventMentionSpan [249,252] [war]
[WARN] Boundary mismatch found in bolt-eng-DF-201-185516-356700: Token [2243,2256] [murdered.html] vs. EventMentionSpan [2243,2251] [murdered]
[WARN] Boundary mismatch found in bolt-eng-DF-201-185516-356700: Token [4803,4816] [murdered.html] vs. EventMentionSpan [4803,4811] [murdered]
[INFO] Successfully completed.
Converting to TBF format
[ERROR] 2015-08-09 22:31:04,287 : Incorrect number of fields in a text bound annotation, the process will try to continue but you should check the ann input.
[ERROR] 2015-08-09 22:31:04,287 : ['T2', 'Justice_Charge-Indict 562 569']
[ERROR] 2015-08-09 22:31:04,287 : Incorrect number of fields in a text bound annotation, the process will try to continue but you should check the ann input.
[ERROR] 2015-08-09 22:31:04,287 : ['T7', 'Conflict_Attack 586 593']
[ERROR] 2015-08-09 22:31:04,315 : Incorrect number of fields in a text bound annotation, the process will try to continue but you should check the ann input.
[ERROR] 2015-08-09 22:31:04,315 : ['T3', 'Justice_Charge-Indict 242 249']
[INFO] 2015-08-09 22:31:12,838 : Finish converting 158 files
```

I'm not sure if this is ok.

And another issue: the 'gold.tbf' file seems a little weird. The token ID list seems to be all the tokens of this document. (The 'gold.tbf' file in your data/conversion_demo directory has the same problem.)

**Comment by hunterhector** [comment]
Hi,

So the root causes of all the problems are actually the upstream XML converter, which might be due to some format change or unknown bug. My colleague is working on that. At the mean time, I tried to provide some workarounds so that at least we can do the conversion. However, it seems that I've created more troubles than solving problems.

I've fixed the all tokens problem. However, it will still give you those warning and errors because the input to brat2tbf is not correct. We will try to work on this as soon as possible.

**Comment by hunterhector** [comment]
Hi, 

So my colleague has found the problem of all the warnings, it is because the new annotation by LDC annotates event mentions within hyperlink tags. When we developed these code, we are notified that such things will not happen.

We have pushed our new implementation which do not ignore hyperlink tags. You will still see the "Boundary mismatch found" warning, however, those can be safely ignored.
